### ISSUE TITLE
PHP warning if a user logs in with his personal token

### ISSUE THREAD

**Issue by btry** [glpi-project/glpi#1534]
Hi

When using the rest API to login with a valid user token : 
```
2017-01-05 22:17:38 [@workstation-01]
  *** PHP Notice(8): Undefined index: glpiactiveentities_string
  Backtrace :
  inc/db.function.php:1714                           
  inc/link.class.php:583                             getEntitiesRestrictRequest()
  inc/search.class.php:5440                          Link::getSearchOptionsToAdd()
  inc/log.class.php:124                              Search::getOptions()
  inc/commondbtm.class.php:381                       Log::constructHistory()
  inc/commondbtm.class.php:1146                      CommonDBTM->updateInDB()
  inc/user.class.php:3733                            CommonDBTM->update()
  inc/auth.class.php:737                             User::manageDeletedUserInLdap()
  inc/api.class.php:177                              Auth->Login()
  inc/apirest.class.php:114                          API->initSession()
  apirest.php:47                                     APIRest->call()
```

First analysis : 
* Auth::login() runs with $noauto == true
* the method sets $user_deleted_ldap to true
* the method checks $user_deleted_ldap and if true, calls User::manageDeletedUserInLdap()
* User::manageDeletedUserInLdap() tries to update the history of the user
* Log::constructHistory() needs $_SESSION['glpiactiveentities_string'] which is not set yet.

Seens to not happen in GLPi 9.1.1, and happen with 9.1/bugfixes (HEAD at time of writing this issue) and GLPi 9.2

unit tests of Flyve MDM  

GLPi 9.1.1 with some backports
https://travis-ci.org/flyve-mdm/flyve-mdm-glpi/jobs/189224844

GLPi 9.1.2
https://travis-ci.org/flyve-mdm/flyve-mdm-glpi/jobs/189224845

**Comment by btry** [comment]
The warning seems happen since this commit https://github.com/glpi-project/glpi/commit/2fab4298f28a1f60a9023a03e4f05c078df34eea

but I also found Auth::login()) sets is_deleted_ldap to 1 and re-updates the user and accidentally resets this flag.

the bug cancels itself, but there is a very short period of time when users are marked as LDAP-deleted when they don't come from a directory.

I think the following should not run 
https://github.com/glpi-project/glpi/blob/master/inc/auth.class.php#L735

and it does because of this 
https://github.com/glpi-project/glpi/blob/master/inc/auth.class.php#L647

**Comment by remicollet** [comment]
I think fix should be in Link::getSearchOptionsToAdd()

**Comment by trasher** [comment]
Fixed from PR #1627
